aws/ec2_transit_gateway.go

```diff
@@ -109,8 +110,11 @@ func ec2DescribeTransitGatewayRoute(conn *ec2.EC2, transitGatewayRouteTableID, d
 		if route == nil {
 			continue
 		}
-
-		if aws.StringValue(route.DestinationCidrBlock) == destination {
+		if cidrBlocksEqual(aws.StringValue(route.DestinationCidrBlock), destination) {
+			// normalize IPv6 to RFC1924 compressed
+			_, ipnet, _ := net.ParseCIDR(aws.StringValue(route.DestinationCidrBlock))
+			cidrString := ipnet.String()
+			route.DestinationCidrBlock = &cidrString
```

Comment:
I hit a similar issue over in #14543 and ended up adding a new function (I put it in `validators.go` to keep it close to the other CIDR functions):

```go
// canonicalCidrBlock returns the canonical representation of a CIDR block.
// This function is especially useful for hash functions for sets which include IPv6 CIDR blocks.
func canonicalCidrBlock(cidr string) string {
	_, ipnet, err := net.ParseCIDR(cidr)
	if err != nil {
		return cidr
	}

	return ipnet.String()
}
```

Reply:
I like this. Looks like your PR isn’t ready yet? Mind if I steal this snippet? It’s cleaner than what I did.

Reply:
Sure, please do.